lib/govspeak.rb

```diff
@@ -138,7 +138,7 @@ def preprocess(source)
     def footnote_definitions(source)
       is_legislative_list = source.scan(/\$LegislativeList.*?\[\^\d\]*.*?\$EndLegislativeList/m).size.positive?
       is_cta = source.scan(/\$CTA.*?\[\^\d\]*.*?\$CTA/m).size.positive?
-      footnotes = source.scan(/\[\^(\d+)\]:(.*)/)
+      footnotes = source.scan(/\[\^(\d+)\]:(.*)/).select { |footnote| footnote[-1].present? }
```

Comment:
I'm a bit confused by how this fixes the problem - the code looks rather cryptic. Would you be able to explain more in the commit message? I'm also not sure I fully understand what the cause of the problem is.

Are you using footnote[-1] as a means to access the last item in an arrays match? if so would `footnote.last` be less cryptic?

I'm also wondering will this bug not still exist if they had any extra characters on a legislative list line. For example:

```
    $LegislativeList
    1.  some text[^1]: extra
    $EndLegislativeList
    [^1]: footnote text
```

or even just some spaces?

Perhaps we should only be matching a footnote definition if there is only whitespace before one? for example `/^\s*\[\^(\d+)\]:(.*)/`

Reply:
Correcting regex is a better solution, thank you @kevindew I have added this additional test to make sure it works as expected.